src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -136,11 +137,19 @@ public SnowflakeConnectionV1(String url, Properties info, boolean fakeConnection
 
   private void initConnectionWithImpl(
       SFConnectionHandler sfConnectionHandler, String url, Properties info) throws SQLException {
+    Stopwatch stopwatch = new Stopwatch();
```

Comment:
do you think we can introduce Stopwatch with lambda to execute between start and stop? e.g.
```
stopwatch.measureTime(() -> {
    logger.info("Initializing new connection");
    this.sfConnectionHandler = sfConnectionHandler;
    sfConnectionHandler.initializeConnection(url, info);
    this.sfSession = sfConnectionHandler.getSFSession();
    missingProperties = sfSession.checkProperties();
    this.showStatementParameters = sfSession.getPreparedStatementLogging();
})
```

Reply:
We could add such lambda, but it may be difficult to log the parameters which are initialized within the lambda in a single line. We would need to have two lines of logs - one at the end of the lambda describing the state of the objects and the other one logging the time of lambda completion.